plugins/channel-safely/channel-safely-plugin.cpp

```diff
@@ -0,0 +1,284 @@
+                                     "\n"));
+    return CR_OK;
+}
+
+DFhackCExport command_result plugin_shutdown(color_ostream &out) {
+    namespace EM = EventManager;
+    EM::unregisterAll(plugin_self);
+    enabled = false;
+    return CR_OK;
+}
+
+command_result manage_channel_designations(color_ostream &out, std::vector<std::string> &parameters) {
+#ifdef CS_DEBUG
+    debug_out = &out;
+#endif
+    namespace EM = EventManager;
+    using namespace EM::EventType;
+    if (debug_out) debug_out->print("manage_channel_designations()\n");
+    // manually trigger managing designations
+    if (parameters.empty()) {
+        if (debug_out) debug_out->print("mcd->manage_designations()\n");
+        ChannelManager::Get().manage_designations(out);
+        if (!enabled) {
+            ChannelManager::Get().delete_groups();
+        }
+    }
+    // enable options
+    else if (parameters.size() == 1 && parameters[0] == "enable") {
+        return plugin_enable(out, true);
+    } else if (parameters.size() == 2 && parameters[0] == "enable" && parameters[1] == "cheats" && !cheat_mode) {
+        cheat_mode = true;
+        out.print("channel-safely: cheat mode enabled!\n");
+    } else if (parameters.size() == 2 && parameters[0] == "enable" && parameters[1] == "ontick" && !useTicks) {
+        useTicks = true;
+        out.print("channel-safely: ontick events enabled!\n");
+        if (enabled) {
+            EM::EventHandler tickHandler(onNewEvent, maxTickFreq);
+            if (!has_when<EM::EventHandler>::value) {
+                EM::registerTick(tickHandler, maxTickFreq, plugin_self);
+            } else {
+                EM::registerListener(EventType::TICK, tickHandler, plugin_self);
+            }
+        }
+    }
+    // disable options
+    else if (parameters.size() == 1 && parameters[0] == "disable") {
+        return plugin_enable(out, false);
+    } else if (parameters.size() == 2 && parameters[0] == "disable" && parameters[1] == "cheats" && cheat_mode) {
+        cheat_mode = false;
+        out.print("channel-safely: ontick events disabled!\n");
+    } else if (parameters.size() == 2 && parameters[0] == "disable" && parameters[1] == "ontick" && useTicks) {
+        useTicks = false;
+        out.print("channel-safely: ontick events disabled!\n");
+        if (enabled) {
+            EM::unregisterAll(plugin_self);
+            EM::EventHandler jobStartHandler(onStart, 0);
+            EM::EventHandler jobCompletionHandler(onComplete, 0);
+            EM::registerListener(EventType::JOB_INITIATED, jobStartHandler, plugin_self);
+            EM::registerListener(EventType::JOB_COMPLETED, jobCompletionHandler, plugin_self);
+        }
+    }
+    // developer debug
+    else if (parameters.size() == 1 && parameters[0] == "debug") {
+        debug_out = &out;
+        ChannelManager::Get().debug();
+        debug_out = nullptr;
+    } else {
+        debug_out = nullptr;
```

Comment:
Suggestion: this could also print out the help message (as-is, `channel-safely` and `channel-safely help` both print nothing - we don't really _require_ plugins to handle those, but it is nice if they can)